tools/cmake/README.md

```diff
@@ -45,21 +45,18 @@ Prerequisities:
 - CMake >=3.19.0
 - mbed-tools >=3.4.0
 
-From the application root or wherever `mbed-os.lib` is found:
-1. Run the following command to create the Mbed OS configuration CMake module: 
+From the application root or wherever `mbed-os.lib` is found, run the following command to:
+ * create the Mbed OS configuration CMake module
+ * create a build directory
+ * generate the project configuration
+ * build the project using the `Ninja` build system
 
     ```
-    mbedtools configure -m <mbed-target> -t <toolchain>
+    mbedtools build -m <mbed-target> -t <toolchain>
     ```
 
-1. Run the following command to create a build directory, generate the project configuration and build the project using `Ninja`: 
-
-    ```
-    mkdir build && cd build && cmake .. -GNinja && cmake --build .
-    ```
-
-The default build type is `Develop`. Use `CMAKE_BUILD_TYPE` to select `Develop`, `Release` or `Debug` as follows: 
```

Comment:
is it possible to keep the "expert" instructions somewhere? 

I like the idea of having to run mbedtools only once and then rely on the output in Git to build the rest. This way I don't have to install mbedtools on my CI machine for example.

Reply:
plus the big +1 for using CMake is to actually be able to use CMake ;)

Reply:
Surely experts know how to use CMake and do not need to be given guidance how to use it. For instructions on how to use CMake one might just need to read CMake documentation which will be more up to date than whatever we will write here.

Reply:
> Surely experts know how to use CMake and do not need to be given guidance how to use it. For instructions on how to use CMake one might just need to read CMake documentation which will be more up to date than whatever we will write here.

@hugueskamba sure, you're right, I meant it more in a way of "let casual user know there's more to it and promote learning of new tools/methodologies/concepts/etc.". I'm a big fan of "this is the usual simpler way, but if you need more control, don't forget you can use the full power of CMake with ... and find more documentation here!"

> Mbed-tools is a requirement to using CMake with Mbed OS as it parses the various Mbed configuration file. So one will always have to download it when working with Mbed. Unless you are not planning on modifying your configurations (including build profile)...

Working on a real life product, hardware specs never change you're far enough in prototyping and even more when you've reached production. 

The product evolves but configuration is not changed unless it's absolutely needed. 

I haven't moved to mbed officiel cmake in production yet, we are still using mbed-cmake, so admittedly what we are doing now might change when we move.

Reply:
Given we override `CMAKE_BUILD_TYPE` with "mbed specific" values (debug, develop etc), shouldn't we at least add a note about that for users who want to run cmake directly? `mbed-tools` will pass `-DCMAKE_BUILD_TYPE=...` for you, but if you're running cmake directly you may need to pass it in yourself i.e `cmake -S <source-dir> -B <build-dir> -DCMAKE_BUILD_TYPE=debug`

Reply:
(Please review)[https://github.com/ARMmbed/mbed-os/compare/ad7be28742e4d18a517b25829d3fdf0f80f3ce64..6abb984473843e5d8ad95b77187c1005b6e3723b].